sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -81,6 +85,27 @@ private RestProxy(HttpPipeline httpPipeline, SerializerAdapter serializer, Swagg
         this.interfaceParser = interfaceParser;
         this.decoder = new HttpResponseDecoder(this.serializer);
         this.responseConstructorsCache = new ResponseConstructorsCache();
+        this.tracerSpanAttributes = null;
```

Comment:
call the other constructor from here instead of duplicating initialization code in both constructors.